src/net/tcp/connection_states.cpp

```diff
@@ -1061,6 +1070,7 @@ State::Result Connection::FinWait1::handle(Connection& tcp, Packet_view& in) {
     if(in.ack() == tcp.tcb().SND.NXT) {
       // TODO: I guess or FIN is ACK'ed..?
       tcp.set_state(TimeWait::instance());
```

Comment:
Maybe make a private helper function on Connection:
```
void enter_timewait()
{
  set_state(State::TimeWait::instance());
  release_memory();
  timewait_start();
}
```
and keep the `rtx_stop()` in those places that have it (order doesn't matter).
Also, the `if(tcp.rtx_timer.is_running())` is redundant since this check is already done in `Timer::stop()`.